transform_cleanup.ml

```diff
@@ -0,0 +1,69 @@
+open Instr
+open Transform_utils
+
+let remove_jmp (instrs : instruction_stream) : instruction_stream option =
+  let pred = Analysis.predecessors instrs in
+  let succ = Analysis.successors instrs in
+  let transform pc =
+    if (pc+1) = Array.length instrs then Unchanged else
+    match[@warning "-4"] instrs.(pc), instrs.(pc+1) with
+    | Goto l1, Label l2 when l1 = l2 && List.length pred.(pc+1) = 1 ->
+      Remove 2
+    | Label _, _ when pred.(pc) = [pc-1] && succ.(pc-1) = [pc] ->
+        (* A label is unused if the previous instruction is the only predecessor
+         * unless the previous instruction jumps to it. The later can happen
+         * if its a goto (then we already remove it -- see above) or if its a branch (which
+         * is excluded by the second tests "succ (pc-1) = [pc]")
+         * TODO: we should implement some generic api for instructions like, "Instr.is_jmp" *)
+      Remove 1
+    | i, _ ->
+      Unchanged
+  in
+  change_instrs transform instrs
+
+let remove_unreachable_code (instrs : instruction_stream) : instruction_stream option =
+  let unreachable =
```

Comment:
I'm only seeing this now and it's unrelated to the current commit, but calling this `reached` or `reachable` makes more sense than `unreachable`.